docs/spec/spec.md

```diff
@@ -944,25 +946,49 @@ response when returning `anydata` directly from a resource method.
 | HEAD              | Retrieve headers                                              | 200 OK                  |
 | OPTIONS           | Retrieve permitted communication options                      | 200 OK                  |
 
-#### 2.3.6. Introspection resource
+#### 2.3.6. OAS resources
 
-The introspection resource is internally generated for each service and host the openAPI doc can be generated 
-(or retrieved) at runtime when requested from the hosted service itself. In order to get the openAPI doc hosted
-resource path, user can send an OPTIONS request either to one of the resources or the service. The link header
-in the 204 response specifies the location. Then user can send a GET request to the dynamically generated URL in the 
-link header with the relation openapi to get the openAPI definition for the service.
+OAS resources are internally generated for each service and host the generated OpenAPI specification for the service in 
+different formats. In order to access these resources user can send an OPTIONS request either to one of the resources or 
+the service base-path. The link header in the 204 response specifies the location for the OAS resources.
 
 Sample service
 ```ballerina
 import ballerina/http;
+import ballerina/openapi;
 
+@openapi:ServiceInfo {
+    embed: true
+}
 service /hello on new http:Listener(9090) {
     resource function get greeting() returns string {
         return "Hello world";
     }
 }
 ```
 
+Output of OPTIONS call to service base path
+
+```ballerina
+curl -v localhost:9090/hello -X OPTIONS 
+*   Trying ::1...
+* TCP_NODELAY set
+* Connected to localhost (::1) port 9090 (#0)
+> OPTIONS /hello HTTP/1.1
+> Host: localhost:9090
+> User-Agent: curl/7.64.1
+> Accept: */*
+> 
+< HTTP/1.1 204 No Content
+< allow: GET, OPTIONS
+< link: </hello/openapi-doc-dygixywsw>;rel="service-desc", </hello/swagger-ui-dygixywsw>;rel="swagger-ui"
+< server: ballerina/2.0.0-beta.2.1
```

Comment:
I think the outputs need to be updated. May be we can do it in a separate PR.

Reply:
Fixed with [1f5b0b7d0d79e46f4c85d64f44758c8039d890eb](https://github.com/ballerina-platform/module-ballerina-http/pull/2039/commits/1f5b0b7d0d79e46f4c85d64f44758c8039d890eb)